modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java

```diff
@@ -463,6 +441,22 @@ public void updateExchangeRates() throws PortalException {
 			ArrayUtil.toLongArray(commerceCurrencyFinder.getCompanyIds()));
 	}
 
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceRegistration = bundleContext.registerService(
+			PortalInstanceLifecycleListener.class,
+			new PortalInstanceLifecycleListenerImpl(
+				commerceCurrencyLocalService),
+			null);
+	}
+
+	@Deactivate
+	protected void deactivate() {
```

Comment:
@JanisZhang The BaseImpl has a deactivate() method, so you should add `@Override` and `super.deactivate()`.